File: backend/gestion-clientes/src/main/java/com/buenrostroasociados/gestion_clientes/service/auth/AuthService.java
package com.buenrostroasociados.gestion_clientes.service.auth;

import com.buenrostroasociados.gestion_clientes.dto.auth.SigninRequest;
import com.buenrostroasociados.gestion_clientes.dto.auth.SigninResponse;
import com.buenrostroasociados.gestion_clientes.dto.auth.SignupRequest;
import org.springframework.transaction.annotation.Transactional;

public interface AuthService {
    @Transactional
    void signUp(SignupRequest signupRequest);

    SigninResponse signIn(SigninRequest loginRequest);

    /*servce  password reset*/
    void sendPasswordResetLink(String email);
    void resetPassword(String token, String newPassword);

    void logout(String token);
}

File: frontend/app-gestion-clientes/src/services/AuthService.ts
import apiClient from "@/config/apiConfig";

export const authService = {
    async signUp(payload) {
        const response = await apiClient.post("/auth/sign_up", payload);
        return response.data;   //Retorna el JSOn Directamente
    },

    async signIn(payload) {
        const response = await apiClient.post("/auth/sign_in", payload);
        return response.data;
    },

    async requestPasswordReset(email) {
        const response = await apiClient.post("/auth/password-reset-request", null, {
            params: { email },
        });
        return response.data;
    },

    async resetPassword(verificationCode, newPassword) {
        const response = await apiClient.post("/auth/password-reset", null, {
            params: { verificationCode, newPassword },
        });
        return response.data;
    },

    async logout(token) {
        const response = await apiClient.post(
            "/auth/logout",
            {},
            {
                headers: { Authorization: `Bearer ${token}` },
            }
        );
        return response.data;
    },

    async refreshToken(refreshToken) {
        const response = await apiClient.post("/auth/refresh-token", { refreshToken });
        return response.data;
    },
};
